go/libkb/json.go

```diff
@@ -131,9 +153,35 @@ func newJSONFileTransaction(f *JSONFile) (*jsonFileTransaction, error) {
 }
 
 func (f *JSONFile) Save() error {
+	tx, txCreated, err := f.getOrMakeTx()
+	if err != nil {
+		return err
+	}
+	if txCreated {
+		// if Save() created a transaction, then abort it if it
+		// still exists on exit
+		defer func() {
+			if tx != nil {
+				tx.Abort()
+			}
+		}()
+	}
+
```

Comment:
`f.getFilename` should now never return the main file.  maybe we should assert that somehow?